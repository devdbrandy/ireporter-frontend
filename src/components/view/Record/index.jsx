import React from 'react';

const Record = (props) => {
  return (
    <div className="card">
      <div className="record-cover">
        <img src="img/image.png" alt="cover" />
        <div className="overlay" />
        <span className="author">
          By:
          {' '}
          <span className="name">John Doe</span>
        </span>
        <span className="tag tag-intervention">Intervention</span>
      </div>
      <div className="record-body">
        <h4 className="record-title">Record title</h4>
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Et totam, eum asperiores optio saepe hic corrupti perferendis earum ab voluptatem.
        </p>
        <span className="location">
          <i className="fas fa-map-marker-alt" />
          234, surulere
        </span>
      </div>
      <div className="record-status">Resolved</div>
    </div>
  );
};

export default Record;
