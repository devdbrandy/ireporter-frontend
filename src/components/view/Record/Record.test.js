import React from 'react';
import { shallow } from 'enzyme';
import Record from './index';

/**
 * Wrapper for enzyme shallow component
 *
 * @returns {object} The mocked props and component
 */
const setup = () => {
  const props = {
    record: {
      images: [],
      author: {
        firstname: 'john',
        lastname: 'doe',
      },
    },
    openModal: jest.fn(),
  };

  const component = shallow(<Record {...props} />);
  return { props, component };
};

describe('<Record />', () => {
  it('renders without crashing given the required props', () => {
    const { component } = setup();
    expect(component).toMatchSnapshot();
  });
});
