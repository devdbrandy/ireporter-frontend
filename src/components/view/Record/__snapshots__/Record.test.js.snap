// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Record /> renders without crashing given the required props 1`] = `
<div
  className="card"
  onClick={[Function]}
>
  <div
    className="record-cover"
  >
    <img
      alt="cover"
    />
    <div
      className="overlay"
    />
    <span
      className="author"
    >
      By:
       
      <span
        className="name"
      >
        john doe
      </span>
    </span>
    <span
      style={
        Object {
          "bottom": 0,
          "fontSize": "1.2rem",
          "padding": "5px",
          "position": "absolute",
          "right": 0,
        }
      }
    >
      06-05-2019
    </span>
    <span
      className="tag tag-intervention"
    />
  </div>
  <div
    className="record-body"
  >
    <h4
      className="record-title"
    />
    <p />
    <span
      className="location"
    >
      <i
        className="fas fa-map-marker-alt"
      />
    </span>
  </div>
  <div
    className="record-status"
  />
</div>
`;
